qiskit/providers/aer/backends/aerbackend.py

```diff
@@ -313,6 +322,217 @@ def _run(self, qobj, job_id=''):
+        circuit = transpile(circuit, basis_gates=basis_gates, optimization_level=0)
+        return circuit
+
+    def _inline_for_loop_op(self, inst):
+        indexset, loop_parameter, body = inst.params
+
+        self._last_flow_id += 1
+        loop_id = self._last_flow_id
+        loop_name = f'loop_{loop_id}'
+
+        ret = QuantumCircuit()
+        for qr in body.qregs:
+            ret.add_register(qr)
+        for cr in body.cregs:
+            ret.add_register(cr)
+        if len(ret.clbits) == 0:
+            ret.add_bits(body.clbits)
+        if len(ret.qubits) == 0:
+            ret.add_bits(body.qubits)
+
+        inlined_body = None
+        break_label = f'{loop_name}_end'
+        for index in indexset:
+            continue_label = f'{loop_name}_{index}'
+            inlined_body = self._inline_circuit(body,
+                                                continue_label,
+                                                break_label)
+            inlined_body = inlined_body.bind_parameters({loop_parameter: index})
+            ret.append(inlined_body,
+                       range(inlined_body.num_qubits),
+                       range(inlined_body.num_clbits))
+            ret.append(AerMark(continue_label, inlined_body.num_qubits),
+                       range(inlined_body.num_qubits), [])
+
+        if inlined_body:
+            ret.append(AerMark(break_label, inlined_body.num_qubits),
+                       range(inlined_body.num_qubits), [])
+        return ret
+
+    def _inline_while_loop_op(self, inst):
+        condition_tuple = inst.condition
+        body, = inst.params
+
+        self._last_flow_id += 1
+        loop_id = self._last_flow_id
+        loop_name = f'while_{loop_id}'
+
+        ret = QuantumCircuit()
+        for qr in body.qregs:
+            ret.add_register(qr)
+        for cr in body.cregs:
+            ret.add_register(cr)
+        if len(ret.clbits) == 0:
+            ret.add_bits(body.clbits)
+        if len(ret.qubits) == 0:
+            ret.add_bits(body.qubits)
+
+        continue_label = f'{loop_name}_continue'
+        loop_start_label = f'{loop_name}_start'
+        break_label = f'{loop_name}_end'
+        inlined_body = self._inline_circuit(body, continue_label, break_label)
+        ret.append(inlined_body,
+                   range(inlined_body.num_qubits),
+                   range(inlined_body.num_clbits))
+        ret.append(AerJump(continue_label, ret.num_qubits),
+                   range(ret.num_qubits), [])
+        ret.append(AerMark(break_label, inlined_body.num_qubits),
+                   range(inlined_body.num_qubits), [])
+        return condition_tuple, continue_label, loop_start_label, break_label, ret
+
+    def _inline_if_else_op(self, inst, continue_label, break_label):
+        condition_tuple = inst.condition
+        true_body, false_body = inst.params
+
+        self._last_flow_id += 1
```

Comment:
This should be passed as a variable to the function, and returned from function if it needs to be updated so you aren't tracking some internal state of a function call for a single circuit in the simulator class